Plugin/PaymentVaultDeleteToken.php

```diff
@@ -54,7 +54,9 @@ public function beforeDelete(
         \Magento\Vault\Api\PaymentTokenRepositoryInterface $subject,
         PaymentTokenInterface $paymentToken
     ) {
-        if (strpos($paymentToken->getPaymentMethodCode(), 'adyen_') !== 0) {
+        $paymentMethodCode = $paymentToken->getPaymentMethodCode();
+
+        if (!is_null($paymentMethodCode) && strpos($paymentMethodCode, 'adyen_') !== 0) {
```

Comment:
When there is no payment method or it's not Adyen we can return early: 
```suggestion
        if (is_null($paymentMethodCode) || strpos($paymentMethodCode, 'adyen_') !== 0) {
```